doc/architectural/cbmc-guide.md

```diff
@@ -1,5 +1,5 @@
 \ingroup module_hidden
-\page cbmc-guide CBMC Guide
+\page cbmc-guide CBMC Developer Manual
```

Comment:
If the document title has changed (which I approve of) is it worth changing the file name as well to match?